neo/Consensus/ConsensusContext.cs

```diff
@@ -17,7 +17,7 @@ namespace Neo.Consensus
 {
     internal class ConsensusContext : IConsensusContext
     {
-        public StateRootBase ProposalStateRoot { get; set; }
+        private StateRoot PreviousBlockStateRoot;
```

Comment:
As expected, we are dealing with a relay of the `PreviousBlockStateRoot`, right @KickSeason?
This rename looks good.

In this sense, there is not a problem in relaying it together with `Preparations` since `Persist` should be done in order to validate a block proposal.